### PR TITLE
KVM release 14.1.0

### PR DESCRIPTION
**Pull request by yulianedyalkova**
Towards https://github.com/giantswarm/giantswarm/issues/10713

### CODE REVIEW

**Comment by yulianedyalkova** [comment]
CNCF passed: https://s3.console.aws.amazon.com/s3/object/automated-test-results.giantswarm.io/745ce196-c275-11eb-925d-c2fb0b2915b8//workspace/cluster/sonobuoy-results.tar

File: kvm/requests.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ releases:
           version: ">= 1.0.1"
           issue: https://github.com/giantswarm/giantswarm/issues/14205
         - name: cluster-operator
-          version: ">= 0.24.2 < 0.25.0"
+          version: ">= 0.24.2"
```
Comment (yulianedyalkova) [inline kvm/requests.yaml:11]: Upgrading app-operator too so we don't need this limitation anymore: https://github.com/giantswarm/releases/pull/650#discussion_r597052466

File: kvm/v14.1.0/release.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  - name: etcd
+    version: 3.4.13
+  - name: flannel-operator
+    version: 0.2.0
+  - name: kubernetes
+    version: 1.19.9
+  - catalog: control-plane-test-catalog
+    name: kvm-operator
+    releaseOperatorDeploy: true
+    reference: 3.17.0-5c4ff71eb6c631a885cd58ebc41eb2de401c944c
```
Comment (yulianedyalkova) [inline kvm/v14.1.0/release.yaml:51]: Obviously I need to release kvm-operator and pin a final version here.

**Comment by yulianedyalkova** [comment]
@giantswarm/team-rocket in the beta release I also see the following changes:
* `cert-exporter`: `1.3.0`->`1.6.1`
* `chart-operator`: `2.12.0`->`2.15.0`
* `kube-state-metrics`: `1.3.0`->`1.3.1`
* `metrics-server`: `1.1.1@0.3.6`->`1.3.0@0.4.1`
* `node-exporter`: `1.7.0`->`1.7.2`
* `containerlinux`: `2512.5.0`->`2765.2.4`
* `etcd`: `3.4.13`->`3.4.16`
* `kubernetes`: `1.19.9`->`1.19.11`

Do we want to/have to include any of them here? The only one that sounds critical is `chart-operator` to add the proxy support.

**Comment by tfussell** [comment]
> @giantswarm/team-rocket in the beta release I also see the following changes:
> 
> * `cert-exporter`: `1.3.0`->`1.6.1`
> * `chart-operator`: `2.12.0`->`2.15.0`
> * `kube-state-metrics`: `1.3.0`->`1.3.1`
> * `metrics-server`: `1.1.1@0.3.6`->`1.3.0@0.4.1`
> * `node-exporter`: `1.7.0`->`1.7.2`
> * `containerlinux`: `2512.5.0`->`2765.2.4`
> * `etcd`: `3.4.13`->`3.4.16`
> * `kubernetes`: `1.19.9`->`1.19.11`
> 
> Do we want to/have to include any of them here? The only one that sounds critical is `chart-operator` to add the proxy support.

Because we're not under a deadline with this release (that I know of) and we've neglected updating any components for quite a long time, I'd prefer to update everything. These have already passed CNCF conformance in the beta release so all that remains is manual checks (pod logs, cluster upgrades, MC and WC node system logs). Flatcar in particular feels important to me and we generally try to stay on the latest Kubernetes patch release if nothing else.

**Comment by tfussell** [comment]
It would be nice to get https://github.com/giantswarm/kvm-operator/pull/1221 in this release (I'm sorry, I know you just released kvm-operator). I'm fine with wiring it in a new patch WC release if you want to go ahead and get this out though.

**Comment by yulianedyalkova** [comment]
>Because we're not under a deadline with this release (that I know of) and we've neglected updating any components for quite a long time, I'd prefer to update everything.

That's a really good point, upgrading everything then. 

>It would be nice to get giantswarm/kvm-operator#1221 in this release (I'm sorry, I know you just released kvm-operator). I'm fine with wiring it in a new patch WC release if you want to go ahead and get this out though.

I'll start testing the other changes but I will wait for that to get merged and released. As you said, we're not in a rush and there is no need to repeat all the work in 2 days again.

File: kvm/v14.1.0/release.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    name: metrics-server
+    version: 1.3.0
+  - name: net-exporter
+    version: 1.10.1
+  - componentVersion: 1.0.1
+    name: node-exporter
+    version: 1.7.2
+  components:
+  - name: app-operator
+    releaseOperatorDeploy: true
```
Comment (rossf7) [inline kvm/v14.1.0/release.yaml:30]:
```suggestion
    releaseOperatorDeploy: false
```

**Review by rossf7 — DISMISSED** [review]
@yulianedyalkova We need to change this flag because the app CR should be managed by cluster-operator.

I've already edited the release CR on beaver as @webwurst got paged for the failed app.

**Review by tfussell — APPROVED** [review]

**Pull request merged**